dbg.h

```diff
@@ -189,12 +189,22 @@ struct has_begin_end_size {
                                 is_detected<detect_size_t, T>::value;
 };
 
+template <typename T>
+using ostream_operator_t = decltype(std::declval<std::ostream&>() << std::declval<T>());
+
+template <typename T>
+struct has_ostream_operator {
+  static constexpr bool value = is_detected<ostream_operator_t, T>::value;
+};
+
+
 // Specializations of "prettyPrint"
 
 template <typename T>
 typename std::enable_if<!has_begin_end_size<T>::value, bool>::type prettyPrint(
     std::ostream& stream,
     const T& value) {
+  static_assert(has_ostream_operator<T>::value, "Type does not support the << ostream operator");
```

Comment:
should this maybe be `has_ostream_operator<const T&>::value`? Because `const T&` is the type of value that is supplied to `stream << ` below.

Reply:
Yes, that is a good idea.